ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanDescriptorManager.java

```diff
@@ -1294,7 +1297,39 @@ private void checkMappedByManyToMany(DeployBeanPropertyAssocMany<?> prop) {
   }
 
   private PlatformIdGenerator createSequenceIdGenerator(String seqName, int stepSize) {
-    return databasePlatform.createSequenceIdGenerator(backgroundExecutor, dataSource, stepSize, seqName);
+    return new PlatformIdGenerator() {
+
+      private Map<DataSource, PlatformIdGenerator> map = Collections.synchronizedMap(new WeakHashMap<>());
+
+      private PlatformIdGenerator create() {
+        return databasePlatform.createSequenceIdGenerator(backgroundExecutor, dataSourceSupplier.getDataSource(), stepSize, seqName);
+      }
+
+      private PlatformIdGenerator get() {
+        return map.computeIfAbsent(dataSourceSupplier.getDataSource(), k -> create());
```

Comment:
FI: This works good for a DS-supplier that supplies different dataSources for different tenants (As it is done in TenantMode.DB/DB_WITH_MASTER - which is our use case @foconis) - for other tenant modes like SCHEMA or CATALOG it may cause other problems. Maybe in a schema/catalog environment this should somehow configurable, where the Sequence lives.